pbs/usersync.go

```diff
@@ -317,9 +319,25 @@ func getRawQueryMap(query string) map[string]string {
 
 func (deps *UserSyncDeps) SetUID(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
 	pc := ParsePBSCookieFromRequest(r, &deps.HostCookieSettings.OptOutCookie)
+	var so analytics.SetUIDObject
```

Comment:
I think you might mean for the to be `var so *analytics.SetUIDObject`

The "zero" value for a `struct` is still a real `struct`... so you're actually making a new object here, and then overwriting it with another new one `if Analytics != nil`.